### ISSUE TITLE
bound generic classes used dynamically?

### ISSUE THREAD

**Issue by kjx** [gracelang/language#40]
Say we have 
`
class BinaryTree<X>  where X <: Comparable {
    method add(e : X) { ... }
 `
which we instantiate and use dynamically:

'
def stupidTree = BinaryTree // implicit dynamic
stupidTree.add(true)
'

when. and why do we get an error that 'true' is not 'Comparable' ?

**Comment by apblack** [comment]
The question that you asked doesn't make sense.  `true` is a value, `Comparable<T>` a property of a type.  I think that you mean `Boolean` is not `Comparable`.

I suspect that we would only get that error if `BinaryTree` were applied to `Boolean` rather than to `Unknown`.  But that's up to those of us who are defining the type dialect.

Dynamically, when the code inside `add` compares `true` with `true` and discovers that there is no method `compare` in `true`, then we get a `NoSuchMethod` error.

**Comment by kjx** [comment]
> The question that you asked doesn't make sense. true is a value, Comparable<T> a property of a type. I think that you mean Boolean is not Comparable.

No, no, I meant true is not an instance of or does not form to or does not match Comparable.
> I suspect that we would only get that error if BinaryTree were applied to Boolean rather than to Unknown.
> 
Yes, we should get that error then. 
> But that's up to those of us who are defining the type dialect
> 
Which needs to be in then main spec...
> Dynamically, when the code inside add compares true with true and discovers that there is no method compare in true, then we get a NoSuchMethod error.
> 
Yep that's what I thought. Here's the thing. Say we start all dynamic:

> class binTree {
>      add(e)

We should get that error when **e** is eventually used. 


Second,  there's the nongeneric typed version: 

> class binTree {
>      add(e : Comparable)

Now if we try to add false well get an immediate error that false is not Comparable. 


Third, we can make it constrained generic:  

> class binTree<T> where T <: Conparable {
>      add(e)

Now - if we instantiate this as binTree<Unknown>,  we'll only get the error later...  Hmm. Do the Contracts people deal with this?  Probably...
